src/test/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetailsTest.java

```diff
@@ -21,4 +25,122 @@ public void testIsFeedbackParticipantCommentsOnResponsesAllowed_shouldReturnFals
         assertFalse(feedbackQuestionDetails.isFeedbackParticipantCommentsOnResponsesAllowed());
     }
 
+    @Test
+    public void tesValidateResponseDetails() {
+        FeedbackNumericalScaleQuestionDetails numScaleQuestion = new FeedbackNumericalScaleQuestionDetails();
+        numScaleQuestion.setStep(0.1);
+
+        ______TS("Test Val=1.2 and Step=0.1 does no trigger error");
+        FeedbackNumericalScaleResponseDetails resp = new FeedbackNumericalScaleResponseDetails();
+        resp.setAnswer(1.2);
+        List<String> errors = numScaleQuestion.validateResponsesDetails(Arrays.asList(resp), 1);
+        assertTrue(errors.isEmpty());
+
+        ______TS("Test Val=1.22 and Step=0.1 triggers error");
+        resp = new FeedbackNumericalScaleResponseDetails();
+        resp.setAnswer(1.22);
+        errors = numScaleQuestion.validateResponsesDetails(Arrays.asList(resp), 1);
+        assertEquals(1, errors.size());
+        assertEquals("Please enter a valid value. The two nearest valid values are 1.2 and 1.3.", errors.get(0));
+
+        ______TS("Test Val=1.333 and Step=0.1 triggers error");
+        resp = new FeedbackNumericalScaleResponseDetails();
+        resp.setAnswer(1.333);
+        errors = numScaleQuestion.validateResponsesDetails(Arrays.asList(resp), 1);
+        assertEquals(1, errors.size());
+        assertEquals("Please enter a valid value. The two nearest valid values are 1.3 and 1.4.", errors.get(0));
+
+        ______TS("Test Val=2 and Step=0.1 does not trigger error");
+        resp = new FeedbackNumericalScaleResponseDetails();
+        resp.setAnswer(2);
+        errors = numScaleQuestion.validateResponsesDetails(Arrays.asList(resp), 1);
+        assertTrue(errors.isEmpty());
+
+        numScaleQuestion.setStep(0.00001);
+
+        ______TS("Test Val=1.33333 and Step=0.00001 does not trigger error");
+        resp = new FeedbackNumericalScaleResponseDetails();
+        resp.setAnswer(1.333);
```

Comment:
Your test description says `1.33333` but the answer here is `1.333`, I'm assuming the description is the correct one?

Reply:
yes updated